mono/mini/memory-access.c

```diff
@@ -16,7 +16,11 @@
 #include "ir-emit.h"
 #include "jit-icalls.h"
 
+#ifdef ENABLE_NETCORE
+#define MAX_INLINE_COPIES 16
```

Comment:
Why 16?

Reply:
I used the number used in Memmove specifically to avoid the recursive call issue. The likely choice for that is that x64 could do that with one unaligned load into xmm register and then one unaligned store from xmm register, effectively moving 16 bytes with two instructions (or multiples of 16 bytes for bigger data types). Mono doesn't do this optimization now but it would be fairly easy to add (and LLVM is already special cased).

Reply:
Here's a link to the CoreCLR limits: https://github.com/dotnet/coreclr/blob/84d4ed44b5c7f1b3db6e9ae3c71e8e1bb585bbaf/src/jit/target.h#L503

Reply:
Does it show any improvements in the microbenchmarks?

Reply:
Yes, it does... and in the generated code that avoids recursive calls. If you are interested in specific number I can trigger the microbenchmark run. Then again, this is not primarily a performance fix as much as it is a correctness fix for the code that depends on this inlining behavior and which is located in the shared CoreLib partition.